### DISCUSSION TITLE
Enabling signatures

### DISCUSSION THREAD

**Discussion by s1037989** [mojolicious/mojo#1632] [Ideas]
What about defaulting to signatures enabled in perhaps the next major release, and provide a -no-signatures option to disable signatures?

**Comment by jberger** [comment]
We've talked at times about doing this but it really can't be until signatures would go stable in the language. Even then there would be some consideration about the interaction between signatures and attributes which changed in certain versions of perl; maybe we'd have to only enable it after a certain perl version. Its going to be interesting.

**Comment by kraih** [comment]
We do still have some functions that use prototypes.
